app/controllers/idv/in_person/state_id_controller.rb

```diff
@@ -31,8 +31,11 @@ def update
           end
 
           analytics.idv_in_person_proofing_state_id_submitted(
-            **analytics_arguments.merge(**form_result.to_h),
+            **analytics_arguments.
+              merge(**form_result.to_h).
+              merge({ birth_year: flow_params.dig(:dob, :year) }),
```

Comment:
this is redundant with returning `FormResponse` so we can remove it right?

Reply:
No. I added `FormResponse` to the `app/services/idv/steps/in_person/state_id_step.rb` which is for the flow state machine. This code is needed for controller side `app/controllers/idv/in_person/state_id_controller.rb` because they are separate code executions.

Reply:
This "duplication" is needed until we migrate away from Flow State Machine.

Reply:
ahh I see now. This uses `Idv::StateIdForm`

We should be adding this logging from inside that form. Here's an example patch, and then we can remove that entirely from here

```diff
diff --git a/app/forms/idv/state_id_form.rb b/app/forms/idv/state_id_form.rb
index c61e6e5fc..5f1b5150f 100644
--- a/app/forms/idv/state_id_form.rb
+++ b/app/forms/idv/state_id_form.rb
@@ -32,6 +32,7 @@ module Idv
       FormResponse.new(
         success: validation_success,
         errors: cleaned_errors,
+        extra: extra_analytics_attributes,
       )
     end
 
@@ -47,5 +48,11 @@ module Idv
     def raise_invalid_state_id_parameter_error(key)
       raise ArgumentError, "#{key} is an invalid state ID attribute"
     end
+
+    def extra_analytics_attributes
+      {
+        birth_year: dob&.year,
+      }
+    end
   end
 end
```

Reply:
[we have an entire workflow around analytics + forms](https://github.com/18F/identity-idp/blob/main/docs/backend.md#forms-formresponse-analytics-and-controllers), it is 100% convention to put things in the FormResponse with the intention of logging them

Reply:
Oh thank you! This is really nice. It also will remove the duplication that was originally thought to be needed. I will work on getting this update in today.